flang/include/flang/Lower/CharacterExpr.h

```diff
@@ -113,8 +113,8 @@ class CharacterExprHelper {
   static fir::CharacterType getCharacterType(mlir::Value str);
 
   /// Return the integer type that must be used to manipulate
-  /// Character lengths. TODO: move this to FirOpBuilder?
-  mlir::Type getLengthType() { return builder.getIndexType(); }
+  /// Character lengths.
+  mlir::Type getLengthType() { return builder.getCharacterLengthType(); }
```

Comment:
Yes, I just moved this choice into the builder because I found it painful to have to use CharacterExprHelper just to get that info when needed outside of CharacterExpr.cpp.